### PR TITLE
Techdebt/use generated document models

### PR DESCRIPTION
**Pull request by FuriousLlama**
Organized the apimodels into different Namespaces

Concepts: Pims related information entities
Requests: Requests and Responses; not information objects but just server requests and responses
Mayan: Mayan model representations
PimsSync: Mayan->Pims Sync.

This is also reflected in the naming of the generated files to avoid confusion

### CODE REVIEW

File: source/frontend/src/components/common/GenericModal.tsx [hunk 1]
```diff
@@ -215,12 +217,19 @@ export const GenericModal = (props: Omit<BsModalProps, 'onHide'> & ModalProps) =
                 title="cancel-modal"
                 variant={cancelButtonVariant ?? 'secondary'}
                 onClick={close}
+                test-id="cancel-modal-button"
```
Comment (asanchezr) [inline source/frontend/src/components/common/GenericModal.tsx:220]: we already have the title property to get the buttons in tests. not opposed to adding test ids but I believe the right syntax is **data-testid="..."** 

See here: https://testing-library.com/docs/queries/bytestid/

File: source/frontend/src/components/common/GenericModal.tsx [hunk 2]
```diff
@@ -215,12 +217,19 @@ export const GenericModal = (props: Omit<BsModalProps, 'onHide'> & ModalProps) =
...
             )}
 
-            <Button title="ok-modal" variant={okButtonVariant ?? 'primary'} onClick={ok}>
+            <Button
+              title="ok-modal"
+              variant={okButtonVariant ?? 'primary'}
+              onClick={ok}
+              disabled={handleOkDisabled}
+              test-id="ok-modal-button"
+            >
```
Comment (asanchezr) [inline source/frontend/src/components/common/GenericModal.tsx:232]: Same comment re: test-id

File: source/frontend/src/components/common/Section/Section.tsx [hunk 3]
```diff
@@ -9,18 +9,29 @@ interface SectionProps {
...
+> = ({
+  header,
+  children,
+  title,
+  isCollapsable,
+  initiallyExpanded,
+  noPadding,
+  className,
+  ...rest
+}) => {
```
Comment (asanchezr) [inline source/frontend/src/components/common/Section/Section.tsx:28]: These (and the modal changes) are in your other PR. They don't look related to this pr. Please make sure to include changes relevant to document model refactoring - thanks!

Comment (FuriousLlama) [reply source/frontend/src/components/common/Section/Section.tsx:28]: these changes are actually related to the document UI changes. Once that pr is merged these will not show on the diff

File: source/backend/api/Areas/DocumentGeneration/DocumentGenerationController.cs [hunk 4]
```diff
@@ -46,7 +47,7 @@ public DocumentGenerationController(IDocumentGenerationService documentGeneratio
 
         // [HasPermission(Permissions.GenerateDocuments)]
         [Produces("application/json")]
-        [ProducesResponseType(typeof(ExternalResult<Models.Cdogs.FileTypes>), 200)]
+        [ProducesResponseType(typeof(ExternalResponse<Models.Cdogs.FileTypes>), 200)]
```
Comment (FuriousLlama) [inline source/backend/api/Areas/DocumentGeneration/DocumentGenerationController.cs:50]: Standardized the naming of backend classes on the API. For 
Result -> Response

**Comment by FuriousLlama** [comment]
Setting it to draft  to            avoid merging it in this release

File: source/backend/api/Areas/Documents/DocumentRelationshipController.cs [hunk 5]
```diff
@@ -5,8 +5,9 @@
 using Microsoft.AspNetCore.Mvc;
 using Pims.Api.Constants;
 using Pims.Api.Helpers.Exceptions;
+using Pims.Api.Models.CodeTypes;
```
Comment (devinleighsmith) [inline source/backend/api/Areas/Documents/DocumentRelationshipController.cs:8]: Do we need both imports?

**Comment by devinleighsmith** [comment]
what is a353046 doing in this PR? it seems unrelated. @FuriousLlama

**Comment by devinleighsmith** [comment]
also 51a9341, seems ui and not model related.

File: source/backend/api/Areas/DocumentGeneration/DocumentGenerationController.cs (hunk 4 above)
Comment (devinleighsmith) [reply source/backend/api/Areas/DocumentGeneration/DocumentGenerationController.cs:50]: I like this, but can we try and separate automate refactoring actions like renames from potentially functional changes like moving files around and actual code changes.

What I would love to see is a PR like this broken out into:
- Automated refactor changes that don't really need a human reviewer.
- Manual changes that could involve human error, like the using changes that broke the lease functionality in the last major backend refactoring PR.

File: source/backend/api/Services/DocumentSyncService.cs [hunk 6]
```diff
@@ -288,8 +291,8 @@ private void AddMayanIdToPimsDocumentTypes(IList<AddDocumentToMayanWithNameRespo
 
         private void SyncDocumentTypeMetadataTypes(SyncModel model, IList<PimsDocumentTyp> pimsDocumentTypes, ref ExternalBatchResult batchResult)
         {
-            Task<ExternalResult<QueryResult<DocumentType>>> documentTypeTask = mayanDocumentRepository.TryGetDocumentTypesAsync(pageSize: 5000);
-            Task<ExternalResult<QueryResult<MetadataType>>> metadataTypesTask = mayanMetadataRepository.TryGetMetadataTypesAsync(pageSize: 5000);
+            Task<ExternalResponse<QueryResponse<Models.Mayan.Document.DocumentTypeModel>>> documentTypeTask = mayanDocumentRepository.TryGetDocumentTypesAsync(pageSize: 5000);
```
Comment (devinleighsmith) [inline source/backend/api/Services/DocumentSyncService.cs:294]: Is there a reason why DocumentTypeModel is fully qualified but MetadataTypeModel isn't?

File: source/backend/api/Services/DocumentSyncService.cs [hunk 7]
```diff
@@ -313,35 +316,35 @@ private void SyncDocumentTypeMetadataTypes(SyncModel model, IList<PimsDocumentTy
             }
         }
 
-        private async Task<ExternalBatchResult> AsyncLinkDocumentMetadata(DocumentTypeModel documentTypeModel, PimsDocumentTyp pimsDocumentTyp, IList<DocumentType> retrievedDocumentTypes, IList<MetadataType> retrievedMetadataTypes)
+        private async Task<ExternalBatchResult> AsyncLinkDocumentMetadata(Models.PimsSync.DocumentTypeModel documentTypeModel, PimsDocumentTyp pimsDocumentTyp, IList<Models.Mayan.Document.DocumentTypeModel> retrievedDocumentTypes, IList<MetadataTypeModel> retrievedMetadataTypes)
```
Comment (devinleighsmith) [inline source/backend/api/Services/DocumentSyncService.cs:319]: oh, is this why? since there are two DocumentTypeModels side-by-side?

File: source/frontend/src/features/documents/ComposedDocument.ts [hunk 8]
```diff
@@ -1,84 +1,99 @@
...
+  statusTypeCode: ApiGen_Base_CodeType<string> | undefined;
   fileName: string | undefined;
   isFileAvailable: boolean | undefined;
   appCreateTimestamp?: UtcIsoDateTime;
   appCreateUserid?: string;
   relationshipId: number | undefined;
-  relationshipType: DocumentRelationshipType | undefined;
-  parentId: number | undefined;
+  relationshipType: ApiGen_CodeTypes_DocumentRelationType | null = null;
+  parentId: string | undefined;
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/ComposedDocument.ts:32]: this doesn't look related to the change to generated types, why is this necessary?

File: source/frontend/src/features/documents/ComposedDocument.ts [hunk 9]
```diff
@@ -1,84 +1,99 @@
...
-        id: document.id,
-        mayanDocumentId: document.mayanDocumentId ?? null,
+        id: document.id || 0,
+        mayanDocumentId: document.mayanDocumentId ?? 0,
         documentType: document.documentType ?? null,
         statusTypeCode: document.statusTypeCode ?? null,
         fileName: document.fileName ?? null,
+        rowVersion: 1,
+
+        appCreateTimestamp: '',
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/ComposedDocument.ts:83]: would it make more sense to change the backend models so these are nullable?

Comment (devinleighsmith) [reply source/frontend/src/features/documents/ComposedDocument.ts:83]: I also wonder if we could somehow make incorporating these common auditing elements more readable? kind of like how we do it on the backend where an underlying mapper handles these audit fields so we don't have to repeat them everywhere.

File: source/frontend/src/features/documents/ComposedDocument.ts [hunk 10]
```diff
@@ -104,21 +119,26 @@ export class DocumentUploadFormData {
...
       metadata.push({
         metadataTypeId: Number(key),
         value: value,
+        id: 0,
       });
     }
 
     return {
-      documentType: documentType,
+      documentTypeId: documentType.id,
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/ComposedDocument.ts:137]: you're pretty confident that these dereferences won't throw NPE's?

File: source/backend/api/Areas/DocumentGeneration/DocumentGenerationController.cs (hunk 4 above)
Comment (FuriousLlama) [reply source/backend/api/Areas/DocumentGeneration/DocumentGenerationController.cs:50]: For the most part, this made the refactor easier on the front end to understand what the model is used for. I agree with splitting it for future prs. unfortunately, I didn't think it would be necessary for this pr.

File: source/frontend/src/features/documents/documentUpload/DocumentUploadModal.tsx [hunk 11]
```diff
@@ -32,6 +32,7 @@ export const DocumentUploadModal: React.FunctionComponent<
   return (
     <GenericModal
       variant="info"
+      className=""
```
Comment (devinleighsmith) [inline source/frontend/src/features/documents/documentUpload/DocumentUploadModal.tsx:35]: With was this necessary?

File: tools/TsModelGenerator/Specifications/PimsGenerationSpec.cs [hunk 12]
```diff
@@ -97,9 +158,44 @@ private InterfaceSpecBuilder ProcessDateTime(InterfaceSpecBuilder builder, Prope
...
+            {
+                builder.MemberName(jsonPropertyAttribute.Name);
+            }
+            return builder;
+        }
+
         public override void OnBeforeBarrelGeneration(OnBeforeBarrelGenerationArgs args)
         {
-            AddBarrel(".", BarrelScope.Files); // adds one barrel file in the global TypeScript output directory containing only files from that directory
+            //AddBarrel(".", BarrelScope.Files); // adds one barrel file in the global TypeScript output directory containing only files from that directory
```
Comment (devinleighsmith) [inline tools/TsModelGenerator/Specifications/PimsGenerationSpec.cs:198]: Can we remove this?

File: tools/TsModelGenerator/Specifications/PimsGenerationSpec.cs [hunk 13]
```diff
@@ -97,9 +158,44 @@ private InterfaceSpecBuilder ProcessDateTime(InterfaceSpecBuilder builder, Prope
             return builder;
         }
 
+        private InterfaceSpecBuilder ProcessDateOnly(InterfaceSpecBuilder builder, PropertyInfo propertyInfo)
+        {
+            // Convert DateOnly to custom date definition
+            if (propertyInfo.PropertyType == typeof(System.DateOnly?) || propertyInfo.PropertyType == typeof(System.DateOnly))
+            {
+                //builder.Type("utcIsoDateOnly", "@/models/api/UtcIsoDateOnly"); Enable when the datetime front end type has been created.
```
Comment (devinleighsmith) [inline tools/TsModelGenerator/Specifications/PimsGenerationSpec.cs:166]: I think this has been created @FuriousLlama

**Review by devinleighsmith — APPROVED** [review]

File: source/backend/api/Services/DocumentSyncService.cs (hunk 7 above)
Comment (FuriousLlama) [reply source/backend/api/Services/DocumentSyncService.cs:319]: thats right. It was kind of ambiguous

File: source/frontend/src/features/documents/ComposedDocument.ts (hunk 9 above)
Comment (FuriousLlama) [reply source/frontend/src/features/documents/ComposedDocument.ts:83]: tbh, I dont think the frontend needs those at all. Probably only some of those are actually needed. i.e create date time, user name
